cardano-constitution/src/Cardano/Constitution/Validator/Dynamic.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TemplateHaskell   #-}
+-- Following is for tx compilation
+{-# LANGUAGE DataKinds         #-}
+{-# OPTIONS_GHC -fplugin PlutusTx.Plugin #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:target-version=1.1.0 #-}
+module Cardano.Constitution.Validator.Dynamic
+    ( constitutionValidator
+    , constitutionValidatorUntyped
+    , mkConstitutionCode
+    , defaultConstitutionValidator
+    , defaultConstitutionValidatorUntyped
+    , defaultConstitutionCode
+    ) where
+
+import Cardano.Constitution.Config
+import Cardano.Constitution.Predicate
+import Cardano.Constitution.ProtocolParam
+import Cardano.Constitution.Validator.Common
+import Cardano.Constitution.Validator.Dynamic.TH
+import PlutusCore.Version (plcVersion110)
+import PlutusLedgerApi.V3 as V3 hiding (ParamName)
+import PlutusTx as Tx
+import PlutusTx.AssocMap as Tx
+import PlutusTx.Prelude as Tx
+
+predicatesMeanings :: Tx.Map PredName PredicateMeaning
+predicatesMeanings = $$(resolvePredicatesMeanings)
+
+lookupMeaningDynamic :: PredName -> PredicateMeaning
+lookupMeaningDynamic predName = case Tx.lookup predName predicatesMeanings of
+                           Nothing -> Tx.error ()
+                           Just m  -> m
+
+-- | Check if a parameter's proposed new value satisfies the constitution config.
+satisfies :: ParamName -> PredName -> ConstitutionConfig -> Tx.Map ParamId PredValue -> Bool
+satisfies paramName predName cCfg cparams
+    | Just expectedPid <- fromParamName paramName cCfg
+    , Just expectedPredVal <- lookupPredValue paramName predName cCfg =
+            case Tx.lookup expectedPid cparams of
+                Just actualPredVal -> (lookupMeaningDynamic predName) expectedPredVal actualPredVal
+                -- This means that the expected parameter remains unchanged *OR*
+                -- the supplied index is unknown
+                Nothing            -> True
+    | otherwise = False
+
+-- Expects a constitution-configuration, statically *OR* at runtime via Tx.liftCode
+constitutionValidator :: ConstitutionConfig -> () -> V3.ScriptContext -> Bool
+constitutionValidator cCfg () =
+    \case
+        -- NOTE: this script will ignore (and thus succeed) on unknown indices inside cparams;
```

Comment:
I was wondering about this.  We should try to get a definitive answer about what's required here, and it should probably be in some specification